ArduSub/motors.cpp

```diff
@@ -130,11 +130,111 @@ void Sub::motors_output()
 {
     // check if we are performing the motor test
     if (ap.motor_test) {
-        return; // Placeholder
+        verify_motor_test();
+    } else {
+        motors.set_interlock(true);
+        motors.output();
     }
-    motors.set_interlock(true);
-    motors.output();
 }
+static uint32_t last_do_motor_test_fail_ms = 0;
+static uint32_t last_do_motor_test_ms = 0;
+// Initialize new style motor test
+// Perform checks to see if it is ok to begin the motor test
+// Returns true if motor test has begun
+bool Sub::init_motor_test()
+{
+    uint32_t tnow = AP_HAL::millis();
+
+    // Ten second cooldown period required with no do_set_motor requests required
+    // after failure.
+    if (tnow < last_do_motor_test_fail_ms + 10000 && last_do_motor_test_fail_ms > 0) {
+        gcs().send_text(MAV_SEVERITY_CRITICAL, "10 second cool down required");
+    }
+
+    // check if safety switch has been pushed
+    if (hal.util->safety_switch_state() == AP_HAL::Util::SAFETY_DISARMED) {
+        gcs().send_text(MAV_SEVERITY_CRITICAL,"Disarm hardware safety switch before testing motors.");
```

Comment:
Nitpicking: I hate the disarm/arm names for a button, it always confuses me, so maybe this is correct, but it looks strange: if switch is in disarm state ask user to disarm switch?

Reply:
yep.

Reply:
The string is too long, it will be sent as "Disarm hardware safety switch before testing motor" (it's 52 characters instead of 50, ironically it truncates in a reasonable manner)

Reply:
interesting, noted.